passes/techmap/clock_gate.cc

```diff
@@ -0,0 +1,205 @@
+#include "kernel/yosys.h"
+#include "kernel/ff.h"
+#include <optional>
+
+USING_YOSYS_NAMESPACE
+PRIVATE_NAMESPACE_BEGIN
+
+struct ClockGateCell {
+	std::string name;
```

Comment:
We can use `IdString` for all the fields here

Reply:
to avoid reconstructing them every time? True I didn't consider that and only wrote what's most ergonomical

Reply:
Not really, it's just slightly more yosys-idiomatic